In [1]:
import pandas as pd

/Users/Monica/.virtualenvs/dataanalysis/lib/python3.5/site-packages/matplotlib/__init__.py:1035: UserWarning: Duplicate key in file "/Users/Monica/.matplotlib/matplotlibrc", line #2
  (fname, cnt))


In [7]:
df = pd.read_csv('data/heights_weights_genders.csv')

In [9]:
print(df.head)

<bound method NDFrame.head of       Gender     Height      Weight
0       Male  73.847017  241.893563
1       Male  68.781904  162.310473
2       Male  74.110105  212.740856
3       Male  71.730978  220.042470
4       Male  69.881796  206.349801
5       Male  67.253016  152.212156
6       Male  68.785081  183.927889
7       Male  68.348516  167.971110
8       Male  67.018950  175.929440
9       Male  63.456494  156.399676
10      Male  71.195382  186.604926
11      Male  71.640805  213.741169
12      Male  64.766329  167.127461
13      Male  69.283070  189.446181
14      Male  69.243732  186.434168
15      Male  67.645620  172.186930
16      Male  72.418317  196.028506
17      Male  63.974326  172.883470
18      Male  69.640060  185.983958
19      Male  67.936005  182.426648
20      Male  67.915050  174.115929
21      Male  69.439440  197.731422
22      Male  66.149132  149.173566
23      Male  75.205974  228.761781
24      Male  67.893196  162.006652
25      Male  68.144033  192.34397

In [10]:
df.corr()

Height    Weight
Height  1.000000  0.924756
Weight  0.924756  1.000000

In [11]:
Male = df[df['Gender'] == 'Male']
Female = df[df['Gender'] == 'Female']

In [12]:
Male.corr()

Height    Weight
Height  1.000000  0.862979
Weight  0.862979  1.000000

In [13]:
Female.corr()

Height    Weight
Height  1.000000  0.849609
Weight  0.849609  1.000000

Findings: For males there was a slightly higher positive correlation between height and weight at 86.29% whereas for females the correlation was 84.96%. Although, I am unclear why prior to  seperating by gender the correlariton was 92% which is higher than both groups as individuals.